locale/locale.c

```diff
@@ -5,17 +5,49 @@
  *
  * locale.c
  *
- * Copyright 2018 Phoenix Systems
+ * Copyright 2018, 2024 Phoenix Systems
  * Author: Michal Miroslaw
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
-#include "locale.h"
+#include <limits.h>
+#include <locale.h>
 
+/* for regex/collate.c */
 char *_PathLocale;
 
-const struct lconv _fixed_locale = {
-	"."
+
+static struct lconv fixed_locale = {
```

Comment:
Hmm, why has this be non-const? If it's because of the usage in `localeconv` (as it sadly returns non-const ptr), then perhaps we might just cast `const` away there?

Reply:
Not the best option of course, so it's ok as is for me

Reply:
I'm a little bit afraid of  user trying to modify the object residing on XIP (rodata) which might trigger flash write (possible waitstates/ realtime breakage). 

With MPU enabled it's not an attack vector (unsafe aplications can be prevented from writing to XIP) so maybe we can make it `const`?

Note: in C99/C11 modifying struct returned from `localeconv` is UB so we can safely throw `MemMgmtFault  :).